app/controllers/api/openid_connect/authorizations_controller.rb

```diff
@@ -0,0 +1,234 @@
+      end
+
+      private
+
+      def reset_auth(auth)
+        return unless auth
+        auth.o_auth_access_tokens.destroy_all
+        auth.id_tokens.destroy_all
+        auth.code_used = false
+        auth.save
+      end
+
+      def handle_prompt(prompt, auth)
+        if prompt.include? "select_account"
+          handle_params_error("account_selection_required",
+                                     "There is no support for choosing among multiple accounts")
+        elsif prompt.include? "none"
+          handle_prompt_none(prompt, auth)
+        elsif prompt.include?("login") && logged_in_before?(60)
+          reauthenticate
+        elsif prompt.include? "consent"
+          request_authorization_consent_form
+        else
+          handle_authorization_form(auth)
+        end
+      end
+
+      def reauthenticate
+        sign_out current_user
+        redirect_to new_api_openid_connect_authorization_path(params)
+      end
+
+      def handle_authorization_form(auth)
+        if auth
+          process_authorization_consent("true")
+        else
+          request_authorization_consent_form
+        end
+      end
+
+      def request_authorization_consent_form
+        add_claims_to_scopes
+        endpoint = Api::OpenidConnect::AuthorizationPoint::EndpointStartPoint.new(current_user)
+        handle_start_point_response(endpoint)
+      end
+
+      def add_claims_to_scopes
+        return unless params[:claims]
+        claims_json = JSON.parse(params[:claims])
+        return unless claims_json
+        claims_array = claims_json["userinfo"].try(:keys)
+        return unless claims_array
+        claims = claims_array.join(" ")
+        req = build_rack_request
+        req.update_param("scope", req[:scope] + " " + claims)
+      end
+
+      def logged_in_before?(seconds)
+        if seconds.nil?
+          false
+        else
+          (Time.zone.now.utc.to_i - current_user.current_sign_in_at.to_i) > seconds.to_i
+        end
+      end
+
+      def handle_prompt_none(prompt, auth)
+        if prompt == ["none"]
+          if auth
+            process_authorization_consent("true")
+          else
+            handle_params_error("interaction_required",
+                                       "The Authentication Request cannot be completed without end-user interaction")
+          end
+        else
+          handle_params_error("invalid_request",
+                                     "The 'none' value cannot be used with any other prompt value")
+        end
+      end
+
+      def handle_start_point_response(endpoint)
+        _status, header, response = *endpoint.call(request.env)
+        if response.redirect?
+          redirect_to header["Location"]
+        else
+          save_params_and_render_consent_form(endpoint)
+        end
+      end
+
+      def save_params_and_render_consent_form(endpoint)
+        @o_auth_application, @response_type, @redirect_uri, @scopes = *[
```

Comment:
Should be able to just drop the `*[...]`, `a, b, c = e, f, g` is just fine. Though I tend to prefer to list them out on their own lines.